### DISCUSSION TITLE
Markdig.Extensions.Footers.FooterBlock Not Implemented. Appears in the changelog comments

### DISCUSSION THREAD

**Discussion by mouzedrift** [ppy/osu#16741] [Q&A] [answered]
osu web:
![brave_1iHJvqU0yL](https://user-images.githubusercontent.com/43358824/152124711-54672853-f0d9-4781-bc44-d63afb5ec570.png)

osu!lazer:
![osu!_OS6KmgtEBE](https://user-images.githubusercontent.com/43358824/152124747-8cadc526-61a3-4405-ad45-0b22477a84a9.png)

on the osu lazer changelog page https://osu.ppy.sh/home/changelog/lazer/2022.127.0

**Comment by smoogipoo** [comment]
Hmm... We're currently just blanket enabling all extensions, but we should probably selectively enable the ones that osu!web uses instead. These are the ones currently enabled, many of which I'm sure aren't handled:
![image](https://user-images.githubusercontent.com/1329837/152126632-b46dcedf-b8de-496e-86fd-f42f4bfff231.png)

@nanaya Is there a list of which markdown extensions osu!web uses, and which flavour of markdown?

**Reply by nanaya** [reply]
different set is enabled for different section

available sets are here

https://github.com/ppy/osu-web/blob/05488a96b25b5a09f2d97c54c06dd2bae59d1dc8/app/Libraries/Markdown/OsuMarkdown.php#L301

section settings are here

https://github.com/ppy/osu-web/blob/05488a96b25b5a09f2d97c54c06dd2bae59d1dc8/app/Libraries/Markdown/OsuMarkdown.php#L36-L135

as for the flavor, it's just commonmark.

**Reply by frenzibyte** [reply]
Seems to translate to the following in MarkDig:
```csharp
        protected override MarkdownPipeline CreateBuilder()
        {
            var pipeline = new MarkdownPipelineBuilder()
                           // corresponds to CommonMark's TablesExtension
                           // - MarkDig's docs: https://github.com/xoofx/markdig/blob/master/src/Markdig.Tests/Specs/GridTableSpecs.md
                           // - CommonMark's docs: https://commonmark.thephpleague.com/2.2/extensions/tables/#syntax
                           .UsePipeTables()
                           // corresponds to CommonMark's StrikethroughExtension
                           .UseEmphasisExtras(EmphasisExtraOptions.Strikethrough)
                           // corresponds to osu-web's parseYamlHeader function.
                           .UseYamlFrontMatter();

            if (Footnotes)
                // corresponds to CommonMark's FootnoteExtension
                pipeline = pipeline.UseFootnotes();

            if (Autolinks)
                // corresponds to CommonMark's AutolinkExtension
                pipeline = pipeline.UseAutoLinks();

            return pipeline.Build();
        }
```

Missing extensions are the special `AttributesExtension`/`AttributesOnlyIdListener` (which I think may be replaced with MarkDig's `AutoIdentifierExtension` or `GenericAttributeExtension`) and the custom osu-web's `StyleBlock/Extension`, both used for wiki. The latter looks to be implementable but I'm not sure how to go through it yet.

Will need to test this as well but can PR if it looks alright.

**Reply by nanaya** [reply]
I just noticed yaml header parser is always active. It should only be for wiki and newsposts...

**Comment by frenzibyte** [comment, marked as answer]
Will be resolved in https://github.com/ppy/osu/pull/20213
